File: CS/DXSample/PageStyleSelector.cs
using System.Windows;
using System.Windows.Controls;

namespace DXSample {
    public class PageStyleSelector : StyleSelector {
        public Style WelcomePageStyle { get; set; }
        public Style CommonPageStyle { get; set; }
        public Style CompletionPageStyle { get; set; }
        public override Style SelectStyle(object item, DependencyObject container) {
            if (!(item is PageViewModel)) return null;
            var vm = item as PageViewModel;
            switch (vm.Type) {
                case PageType.Welcome:
                    return WelcomePageStyle;
                case PageType.Common:
                    return CommonPageStyle;
                case PageType.Completion:
                    return CompletionPageStyle;
            }
            return base.SelectStyle(item, container);
        }
    }
}

File: VB/DXSample/PageStyleSelector.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace DXSample
    Public Class PageStyleSelector
        Inherits StyleSelector

        Public Property WelcomePageStyle() As Style
        Public Property CommonPageStyle() As Style
        Public Property CompletionPageStyle() As Style
        Public Overrides Function SelectStyle(ByVal item As Object, ByVal container As DependencyObject) As Style
            If Not(TypeOf item Is PageViewModel) Then
                Return Nothing
            End If
            Dim vm = TryCast(item, PageViewModel)
            Select Case vm.Type
                Case PageType.Welcome
                    Return WelcomePageStyle
                Case PageType.Common
                    Return CommonPageStyle
                Case PageType.Completion
                    Return CompletionPageStyle
            End Select
            Return MyBase.SelectStyle(item, container)
        End Function
    End Class
End Namespace
